cmds/webboot/webboot.go

```diff
@@ -142,15 +146,25 @@ func (d *DownloadOption) exec(uiEvents <-chan ui.Event, network bool, cacheDir s
 	if err != nil {
 		return nil, err
 	}
-
 	var link string
 	if entry.Label() == customLabel {
 		link, err = menu.PromptTextInput("Enter URL:", validURL, uiEvents)
 		if err != nil {
 			return nil, err
 		}
 	} else {
-		distro := supportedDistros[entry.Label()]
+		//Code for after the specific distro has been selected.
+		//Looks up the distro.
+		distro := supportedDistros[entry.Label()] 
+		if (len(distro.mirrors) > 0) {
+			//Make an array of type menu.Entry to store the mirrors of the 
+			//particular distro selected. Then, display the mirror options.
+			entries := make([]menu.Entry, len(distro.mirrors))
```

Comment:
The user also needs the option to select the URL which is not a mirror. Perhaps the logic for this can be done easily if the default option is the mirror at index 0.

Reply:
I'm looking at types.go and I see just one URL associated with each mirror. Are mirrors going to have multiple URLs in the future? Is that what another menu is needed for? Selecting the particular mirror URL the user wants?